packages/core/core-flows/src/product/workflows/create-product-variants.ts

```diff
@@ -12,9 +16,109 @@ import { createVariantPricingLinkStep } from "../steps/create-variant-pricing-li
 type WorkflowInput = {
   product_variants: (ProductTypes.CreateProductVariantDTO & {
     prices?: PricingTypes.CreateMoneyAmountDTO[]
+  } & {
+    inventory_items?: {
+      inventory_item_id: string
+      required_quantity: number
+    }[]
   })[]
 }
 
+const buildLink = (
+  variant_id: string,
+  inventory_item_id: string,
+  required_quantity?: number
+) => {
+  const link: LinkDefinition = {
+    [Modules.PRODUCT]: { variant_id },
+    [Modules.INVENTORY]: { inventory_item_id: inventory_item_id },
+    data: { required_quantity: required_quantity },
+  }
+
+  if (required_quantity) {
+    link.data = { required_quantity: required_quantity }
+  }
+
+  return link
+}
+
+const buildLinksToCreate = (data: {
+  createdVariants: ProductTypes.ProductVariantDTO[]
+  inventoryIndexMap: Record<number, InventoryNext.InventoryItemDTO>
+  input: WorkflowInput
+}) => {
+  let index = 0
+  const linksToCreate: LinkDefinition[] = []
+
+  for (const variant of data.createdVariants) {
+    const variantInput = data.input.product_variants[index]
+    const shouldManageInventory = variant.manage_inventory
+    const hasInventoryItems = variantInput.inventory_items?.length
+    index += 1
+
+    if (!shouldManageInventory) {
+      continue
+    }
+
+    if (!hasInventoryItems) {
+      const inventoryItem = data.inventoryIndexMap[index]
+
+      linksToCreate.push(buildLink(variant.id, inventoryItem.id))
+
+      continue
+    }
+
+    for (const inventoryInput of variantInput.inventory_items || []) {
+      linksToCreate.push(
+        buildLink(
+          variant.id,
+          inventoryInput.inventory_item_id,
+          inventoryInput.required_quantity
```

Comment:
should we default it to 1 at the application level? even if the db has 1 by default.

Reply:
only on the workflow or the http as well? I like the idea of requiring it on the http, but happy to make this optional and default to 1. wdyt?

Reply:
Yeah, agree. on the http I think we can require it. But the workflow should set it to 1 by default and be optional